Add an optional timezone parameter to /api/calendars

ronshapiro opened the pull request:
Discussions:
  - https://groups.google.com/forum/?fromgroups#!topic/sefaria-dev/r-MjlKAr8z8
  - https://groups.google.com/forum/?fromgroups#!topic/sefaria-dev/P31bOJ0LIcc\

I don't have a local dev environment set up, so I wasn't able to test things, but I tried within a repl to make sure that this was correct. If someone that does have a dev setup could validate, that would be great.

EliezerIsrael commented:
@ronshapiro Thanks for the patch!  It looks good to my eyes.  
I've ask one more person to give it a review and test.

nsantacruz commented on reader/views.py:
```diff
@@ -2199,13 +2200,23 @@ def calendars_api(request):
+            datetimeobj = datetime.datetime(year, month, day, tzinfo=zone)
         except Exception as e:
-            datetimeobj = timezone.localtime(timezone.now())
+            datetimeobj = timezone.localtime(zone)
```
Looks like the `timezone` param needs to be specified since it's the 2nd parameter
```suggestion
            datetimeobj = timezone.localtime(timezone=zone)
```

nsantacruz commented on reader/views.py:
```diff
@@ -2199,13 +2200,23 @@ def calendars_api(request):
+            except pytz.exceptions.UnknownTimeZoneError as e:
+                return jsonResponse({"error": "Unknown 'timezone' value: '%s'." % zone})\
+        else:
+            zone = timezone.now()
```
It seems like `zone` should be of type which is subclass of `tzinfo`. Here you're setting it to `datetime.datetime`. Pretty sure this is an error.

ronshapiro replied:
Hrm, yeah, you're right. I'd like to have the exception case that uses the localtime to also consider the timezone. I think that reverting that line, and then taking your suggestion above should do the trick

nsantacruz commented:
@ronshapiro hi! I've found 2 issues with this pull request. Can you fix those?

ronshapiro commented:
@nsantacruz thanks for the close eyes! I fixed those issues, and I fixed another below where I didn't properly propagate the timezone name to the response if it was specified as a parameter.

Do you prefer I squash my commits, or you'll do that automatically when merging?

nsantacruz commented:
@ronshapiro Looks great! Merged

ronshapiro commented:
Awesome! What's the best way to update the documentation? And how often do
you typically update your servers?

בתאריך יום א׳, 22 במרץ 2020, 14:43, מאת Noah Santacruz ‏<
notifications@github.com>:

> @ronshapiro <https://github.com/ronshapiro> Looks great! Merged
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Sefaria/Sefaria-Project/pull/532#issuecomment-602194379>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAGBRXPVWP3AKZA4P32Z4U3RIYBYFANCNFSM4LPGXKTA>
> .
>
